src/dualscreen.tsx

```diff
@@ -389,10 +389,53 @@ export class DualScreen extends KalturaPlayer.core.BasePlugin implements IEngine
   };
 
   private _getThumbs() {
-    const kalturaCuePointService :any = this._player.getService('kalturaCuepoints');
-    kalturaCuePointService?.registerTypes([kalturaCuePointService.CuepointType.SLIDE]);
+    const kalturaCuePointService: any = this._player.getService('kalturaCuepoints');
+    kalturaCuePointService?.registerTypes([kalturaCuePointService.CuepointType.SLIDE, kalturaCuePointService.CuepointType.VIEW_CHANGE]);
   }
 
+  private _updateSlideLayout = () => {
+    if (this._imagePlayer.active) {
+      this._setMode();
+    }
+  };
+
+  private _onSlideViewChanged = ({playerViewModeId}: ViewChangeData) => {
+    switch (playerViewModeId) {
+      case 'pip-parent-in-large':
+        if (this._singleLayout || this._layout !== Layout.PIP || this._inverse) {
+          this._layout = Layout.PIP;
+          this._inverse = false;
+          this._updateSlideLayout();
+        }
+        break;
+      case 'pip-parent-in-small':
+        if (this._singleLayout || this._layout !== Layout.PIP || !this._inverse) {
+          this._layout = Layout.PIP;
+          this._inverse = true;
+          this._updateSlideLayout();
+        }
+        break;
+      case 'sbs-parent-in-left':
+        if (this._singleLayout || this._layout !== Layout.SideBySide || this._inverse) {
+          this._layout = Layout.SideBySide;
+          this._inverse = false;
+          this._updateSlideLayout();
+        }
+        break;
+      case 'sbs-parent-in-right':
+        if (this._singleLayout || this._layout !== Layout.SideBySide || !this._inverse) {
+          this._layout = Layout.SideBySide;
+          this._inverse = true;
+          this._updateSlideLayout();
```

Comment:
_updateSlideLayout calls _setMode. _setMode already checks if it image and image is not active. So you can just use _setMode.
But I think maybe better is to add this check:
if (this._secondaryPlayerType === PlayerType.IMAGE && !this._imagePlayer.active) return
in the top of switchToPip, SwitchToSidebySide and use them directly in the switch cases. That way it will be much clearer I think